iAM-legacy/Simulation/SplitTreatment.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Simulation.Interface;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Simulation
+{
+    public class SplitTreatment : ISplitTreatment
+    {
+        public string Id { get; }
+
+        public string Description { get; }
+
+        public Criterias Criteria { get; }
+
+        public List<ISplitTreatmentLimit> Limits { get; }
+
+
+        public SplitTreatment(string splitTreamentId, string description, string criteria,  List<ISplitTreatmentLimit> limits)
+        {
+            Id = splitTreamentId;
+            Description = description;
+            Limits = limits;
+            Criteria = new Criterias("SPLIT_TREATMENT", "BINARY_CRITERIA", splitTreamentId);
+            byte[] assemblyCriteria = null;
+            string currentCriteria = "";
+            if (!string.IsNullOrWhiteSpace(criteria)) currentCriteria = Simulation.ConvertOMSAttribute(criteria);
```

Comment:
There appears to be a lot going on in the constructor. I'd consider making some private functions to operate on the data instead, or introduce a new set of classes to do that work and inject them in.

Reply:
Agree, but follows the pattern of about 20 other classes that initialize criteria.  Will refactor all in future.